File: CODE/bin/lengths.pl
#!/usr/bin/env perl


if(length($ARGV[0])<2){
    $ARGV[0] = "goatpox_2_virus_2_2_.fasta"; }

open(IN, $ARGV[0]);
open(OUT, ">./genome.lengths.txt");
@a=split/>/,(join"",<IN>); 
for $e (1..$#a){
	($n,$s)=split/\n/,$a[$e], 2;
	print OUT "$n\t", length($s),"\n"; 
} # end for each sample
exit;

File: bin/lengths.py
#!/usr/bin/env python3

import sys

def calculate_genome_lengths(input_file, output_file="genome.lengths.txt"):
    with open(input_file, "r") as infile, open(output_file, "w") as outfile:
        sequences = infile.read().split(">")
        for seq in sequences[1:]:  # Skip the first empty split
            header, *sequence = seq.split("\n", 1)
            sequence = "".join(sequence).replace("\n", "")
            outfile.write(f"{header}\t{len(sequence)}\n")

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python script.py <input_fasta>")
    else:
        calculate_genome_lengths(sys.argv[1])
